src/inmanta/server/bootloader.py

```diff
@@ -31,15 +69,77 @@ def get_server_slice(self) -> server.Server:
         return server.Server(agent_no_log=self.agent_no_log)
 
     def get_agent_manager_slice(self) -> AgentManager:
-        return AgentManager(self.restserver)
+        return AgentManager()
 
-    def get_server_slices(self) -> List[ServerSlice]:
+    def get_bootstrap_slices(self) -> List[ServerSlice]:
         return [self.get_server_slice(), self.get_agent_manager_slice()]
 
     async def start(self) -> None:
-        for mypart in self.get_server_slices():
+        for mypart in self.load_slices():
```

Comment:
self.restserver.add_slice() doesn't keep track of the order in which the slices were added to the Server.